dts/bindings/usb/zephyr,hid-device.yaml

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Bindings for HID device
+
+compatible: "zephyr,hid-device"
+
+include: base.yaml
+
+properties:
+  interface-name:
+    type: string
+    description: |
+      HID device name. When this property is present, a USB device will use it
+      as the string descriptor of the interface.
+
+  boot-protocol-code:
+    type: string
+    description: |
+      Device can be used as a Boot Device and supports a predefined protocol
+      for mouse or keyboards.
+      - none:     Device does not support the boot interface
+      - keyboard: Device supports boot interface and keyboard protocol
+      - mouse:    Device supports boot interface and mouse protocol
+    enum:
+      - none
```

Comment:
`none` should be removed from here. I know this requires extra macro mapping, but implicitly relying on numbers being the same between different types is something I prefer to avoid.

Reply:
I did not get it. What types are different and how does removing "none" fix "implicitly relying on numbers being the same between different types" ?

Reply:
`none` doesn't really belong under "boot-protocol" from devicetree point of view IMHO (the absence of property signifies that). The "different types" I refer to the enum defined here in devicetree and the protocol code as defined by HID Specification - these two are essentially separate.